src/ActionList/Group.tsx

```diff
@@ -58,32 +79,64 @@ export const Group: React.FC<React.PropsWithChildren<ActionListGroupProps>> = ({
       }}
       {...props}
     >
-      {title && <Header title={title} variant={variant} auxiliaryText={auxiliaryText} labelId={labelId} />}
+      {(slots.groupHeading || title) && (
+        <GroupHeading
+          title={title}
+          variant={variant}
+          auxiliaryText={auxiliaryText}
+          groupHeadingId={groupHeadingId}
+          as={slots.groupHeading?.props.as}
+        >
+          {slots.groupHeading ? slots.groupHeading.props.children : null}
+        </GroupHeading>
```

Comment:
In this case, are `GroupHeading` and `slots.groupHeading` the same component? If so, would `cloneElement` work in this scenario?

```suggestion
        {cloneElement(slots.groupHeading, {
          title,
          variant,
          auxiliaryText,
          groupHeadingId,
        })}
```

Reply:
They are the same components yes.

When I tried cloning though it is complaining about `slots.groupHeading` might be undefined which is a valid case. If ActionList is using the "to-be-deprecated" API which is `<ActionList.Group title="Group heading">` , `slots.groupHeading` is going to be undefined because I use `GroupHeading` component to render both (`title="Group heading"` and `<ActionList.GoupHeading>Group heading</GroupHeading>`),

They are almost the same component except one is taking `title` prop and the other is taking `children` prop to render the heading. I can try splitting them up, or if you have any suggestion I would love to hear it!